jupyterlab_gcedetails/src/components/machine_type_select.tsx

```diff
@@ -0,0 +1,257 @@
+from on the menu */
+
+interface HeaderItemProps {
+  label: string;
+  onHover: () => void;
+}
+
+interface HeaderItemState {
+  isHovering: boolean;
+}
+
+class HeaderItem extends React.Component<HeaderItemProps, HeaderItemState> {
+  constructor(props: HeaderItemProps) {
+    super(props);
+
+    this.state = {
+      isHovering: false,
+    };
+  }
+
+  render() {
+    const { label, onHover } = this.props;
+
+    return (
+      <div onMouseEnter={() => onHover()}>
+        <ListItem button>
+          <ListItemText primary={label} />
+          <ChevronRightRoundedIcon />
+        </ListItem>
+      </div>
+    );
+  }
+}
+
+/* Nested select body component */
+
+interface NestedSelectBodyProps {
+  onSelect: (o: Option) => void;
+  selectedOption: Option;
+}
+
+interface NestedSelectBodyState {
+  displayedOptions: Option[];
+}
+
+class NestedSelectBody extends React.Component<
+  NestedSelectBodyProps,
+  NestedSelectBodyState
+> {
+  constructor(props: NestedSelectBodyProps) {
+    super(props);
+
+    this.state = {
+      displayedOptions: MACHINE_TYPES[0].configurations,
+    };
+  }
+
+  /* Displays a menu of header items that, on hover, change the options the user 
+        can select on the menu */
+  private headersList() {
+    return (
+      <div className={STYLES.itemGroup}>
+        <List dense={true} disablePadding={true}>
+          <HeaderItem
+            key={MACHINE_TYPES[0].base.value}
+            label={MACHINE_TYPES[0].base.text}
+            onHover={() =>
+              this.setState({
+                displayedOptions: MACHINE_TYPES[0].configurations,
+              })
+            }
+          />
+          <Divider style={{ marginTop: 4, marginBottom: 4 }} />
+          {MACHINE_TYPES.slice(1).map(machineType => (
+            <HeaderItem
+              key={machineType.base.value}
+              label={machineType.base.text}
+              onHover={() =>
+                this.setState({ displayedOptions: machineType.configurations })
+              }
+            />
+          ))}
+        </List>
+      </div>
+    );
+  }
+
+  /* Displays a list of options that the user can select from */
+  private optionsList() {
+    const { displayedOptions } = this.state;
+    const { onSelect, selectedOption } = this.props;
+
+    return (
+      <div className={STYLES.itemGroup}>
+        <List dense={true} disablePadding={true}>
+          {displayedOptions.map(o => (
+            <Item
+              key={o.value}
+              option={o}
+              onSelect={onSelect}
+              selected={o.value === selectedOption.value}
+            />
+          ))}
+        </List>
+      </div>
+    );
+  }
+
+  render() {
+    return (
+      <Paper elevation={5} square>
+        <div className={STYLES.menuContainer}>
+          <Grid container spacing={0} alignItems="center">
+            <Grid item xs>
+              {this.headersList}
```

Comment:
Is this meant to reference the method and not actually call it?

Reply:
good catch!